sdk/src/utils/txn-utils.ts

```diff
@@ -38,13 +42,35 @@ export async function checkMergedTransactionSizeIsValid(
     lastValidBlockHeight: number;
   }>
 ): Promise<boolean> {
-  const merged = new TransactionBuilder(ctx.connection, ctx.wallet);
+  const merged = new TransactionBuilder(ctx.connection, ctx.wallet, ctx.txBuilderOpts);
   builders.forEach((builder) => merged.addInstruction(builder.compressIx(true)));
-
   try {
-    const size = await merged.txnSize({ latestBlockhash }); // throws if txnSize is too large
+    const size = await merged.txnSize({
+      latestBlockhash,
+    });
     return true;
   } catch (e) {
     return false;
   }
 }
+
+export function contextToBuilderOptions(
+  opts: WhirlpoolContextOptions
+): TransactionBuilderOptions | undefined {
+  if (opts) {
```

Comment:
nit: is there some possibility that ``opts`` becomes undefined or null ?
I think its definition (``opts: WhirlpoolContextOptions``) doesn't allow it becomes to them.

Reply:
ah good catch. I was attempting to make TransactionBuilderOptions a nested partial and opts would populate it optionally, but Typescript made it quite difficult. I've reverted this to a direct conversion.